### PR TITLE
There wasn't actual persistence, this is fixed

### PR DESCRIPTION
**Pull request by paymog**
Factories are made new for each controller. Services are not.
Persistence is now fixed using a service. Tests pass. We need
integration tests so that this could have been caught the first
time round.

I also did some misellaneous cleanup. Specifically, the controllers
for matched and matching states do not have URLs anymore.

### CODE REVIEW

File: webapp/public/javascripts/services.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+var lunchrServices = angular.module('lunchrServices', []);
+
+lunchrServices.service('authService', function () {
+    var user;
```
Comment (kjrcda) [inline webapp/public/javascripts/services.js]: maybe default assign this to NULL?

File: webapp/public/javascripts/angular-app.js [hunk 2]
```diff
@@ -18,18 +18,18 @@ lunchrApp.config(['$stateProvider', '$urlRouterProvider', '$locationProvider',
                 controller: 'UserController'
             })
             .state('users.matching', {
-                url: '/matching',
```
Comment (kyleleung75) [inline webapp/public/javascripts/angular-app.js:21]: why is it that the url field is now empty? same with line 26

Comment (paymog) [inline webapp/public/javascripts/angular-app.js:21]: It didn't make sense to me to have a URL for this state. By having a URL it means that the user could potentially manually navigate to "/users/matching" even if they hadn't pressed the button to start being matched. Also, the semantics just seemed off having the URL be modified to start being matched and modified again once a match is found.

File: webapp/public/javascripts/services.js (hunk 1 above)
Comment (paymog) [inline webapp/public/javascripts/services.js]: Will do.

File: webapp/public/javascripts/angular-app.js (hunk 2 above)
Comment (kyleleung75) [inline webapp/public/javascripts/angular-app.js:21]: Oh that makes sense, cool.

**Comment by kjrcda** [comment]
looks good

**Comment by kyleleung75** [comment]
It looks good now

**Comment by paymog** [comment]
Lol finally

**Pull request merged**